### ISSUE TITLE
Add Jest for Unit Testing

### ISSUE THREAD

**Issue by pavlovcik** [ubiquity/scraper#6]
This will be used in other repos. Set up Jest to run and report tests.

**Comment by seprintour** [comment]
So we set up jest on the repo code and run it on github when an issue is open?

**Comment by pavlovcik** [comment]
Yeah we need automated tests. This bounty is strictly just to set up jest locally. We can make new bounties later. 

My expectation is that you will need to test basic functionality of this but at the same time I fear that it will take some time to become familiar with how to use this tool. I don't believe i wrote any documentation around it, and it's a pretty complex codebase.

**Comment by seprintour** [comment]
> Yeah we need automated tests. This bounty is strictly just to set up jest locally. We can make new bounties later for CI.
> 
> My expectation is that you will need to test basic functionality of this, like a few tests to check if it can return results from a page, but at the same time I fear that it will take some time to become familiar with how to use this tool. I don't believe I wrote any documentation around it, and it's a pretty complex codebase.
> 
> Starting collaboration with other developers will force me to write documentation which will move things forward in the right direction.

So we'd have go through this issue slowly, shouldn't be too hard for me to get a grasp of.. with your help it'd make it even faster

**Comment by pavlovcik** [comment]
Right now I use this repository as a submodule in a parent "scraper" project. I then import it, and pass in a "pages" directory which dynamically imports logic based on the file structure. For example if scraping GitHub.com then the logic would be in a directory called `GitHub.com/index.ts` 

The interesting part about this feature is that it works arbitrarily deep. For example I could make a file called `GitHub.com/ubiquity/*/discussions/index.ts` and it will automatically be imported when the browser navigates to any of our discussions.

I don't have the time to create an example repository now, but perhaps you can and add me as a direct contributor and we can slowly proceed there.

**Comment by pavlovcik** [comment]
Once the initial jest infrastructure is set up we can make more bounties to write specific types of tests.

**Comment by molecula451** [comment]
Kindly use xDai for issuing after reviewing and merge

**Comment by molecula451** [comment]
> Member

https://github.com/ubiquity/scraper-kernel/pull/10

**Comment by pavlovcik** [comment]
> Kindly use xDai for issuing after reviewing and merge

We only support WXDAI, which is an ERC20 wrapped version.

**Comment by molecula451** [comment]
> Member

absolutely, WXDAI, thank you magnific friend
